mock_helpers_test.go

```diff
@@ -1,25 +1,9 @@
 package autopilot
 
 import (
-	"testing"
-
 	"github.com/hashicorp/raft"
 )
 
-func NewMockTimeProvider(t *testing.T) *mockTimeProvider {
```

Comment:
A new version of mockery was used to generate the mocks. It generates the `New*` methods like we would have here and ensures that expectations are setup using `t.Cleanup`